### Real Estate Customer Interest Analysis: Geographic & Property Insights

#### 1. Objective

This comprehensive analysis investigates customer preferences in various areas and property types in the central of Thailand. This analysis reveals market trends, with insights ranging from urban condominiums to rural homes. This actional insights is to ensure the offers meet the potential customer and growing market needs. The developers, investors, and real estate professionals can make informed decisions, optimize, and tailor their strategies to resonate with market sentiments.

#### 2. Dataset

The dataset, titled “Customer Interest by Location”, is sourced from Bestimate by Baania (https://gobestimate.com/data-detail/Customer-Interest-by-Location). It offers a comprehensive view of customer preferences within the real estate sector in Thailand. The dataset was collected from January 2019 to January 2023.

#### 3. Data Analysis using SQL

#### 3.1 Data Cleaning

- Connect to Postgres database

In [1]:
# load sql extension
%load_ext sql

In [2]:
# connect to database
%sql postgresql://postgres:080437@localhost:5432/customerinterest

- Overview of dataset

In [3]:
%%sql 
select *
from customer_interest
limit 5;

 * postgresql://postgres:***@localhost:5432/customerinterest
5 rows affected.


[('\ufeff1476', '2022', '5', 'Nonthaburi', 'PAK KRET', 'Detached House', 43, 413),
 ('1262', '2022', '10', 'Bangkok', 'DON MUEANG', 'Twin house', 11, 49),
 ('1415', '2020', '4', 'Bangkok', 'PHASI CHAROEN', 'Detached House', 43, 253),
 ('1865', '2020', '1', 'Bangkok', 'POM PRAP SATTRU PHAI', 'Commercial', 25, 75),
 ('1085', '2019', '1', 'Bangkok', 'HUAI KHWANG', 'Condo', 50, 188)]

In [4]:
%%sql 
select count(*) 
from customer_interest;

 * postgresql://postgres:***@localhost:5432/customerinterest
1 rows affected.


[(2358,)]

Observation: Dataset consists of 2358 rows.

- Handle missing values

In [5]:
%%sql
select *
from customer_interest
where transaction_year IS NULL
and province_name_en IS NULL
and district_name_en IS NULL
and propertytype_name_en IS NULL
and volume_of_project IS NULL
and interest IS NULL;

 * postgresql://postgres:***@localhost:5432/customerinterest
0 rows affected.


[]

Observation: No missing values show in the dataset.

- Check the error values

In [6]:
%%sql 
-- 'interest' column should have only positive values
select interest
from customer_interest
where interest < 0;

 * postgresql://postgres:***@localhost:5432/customerinterest
0 rows affected.


[]

- Check and remove duplicate rows

In [7]:
%%sql
select 
    transaction_year,
    province_name_en,
    district_name_en,
    propertytype_name_en,
    volume_of_project,
    interest
from customer_interest
group by 
    transaction_year,
    province_name_en,
    district_name_en,
    propertytype_name_en,
    volume_of_project,
    interest
having count(*) > 1;

 * postgresql://postgres:***@localhost:5432/customerinterest
0 rows affected.


[]

#### 3.2 Data Exploration

1)  Overall customer interest by province, Which province have the highest and lowest customer interest?

    By ranking the customer interest by provinces, the developers can determine the potential market, prioritize investment in those areas, and allocate their marketing efforts.

In [8]:
%%sql
select distinct province_name_en
from customer_interest;

 * postgresql://postgres:***@localhost:5432/customerinterest
6 rows affected.


[('Pathum Thani',),
 ('Samut Prakan',),
 ('Nonthaburi',),
 ('Bangkok',),
 ('Samut Sakhon',),
 ('Nakhon Pathom',)]

In [9]:
%%sql
select
    province_name_en,
    sum(interest) as total_interest
from customer_interest
group by province_name_en
order by total_interest desc;

 * postgresql://postgres:***@localhost:5432/customerinterest
6 rows affected.


[('Bangkok', 543947),
 ('Nonthaburi', 259003),
 ('Pathum Thani', 229066),
 ('Samut Prakan', 223668),
 ('Nakhon Pathom', 110289),
 ('Samut Sakhon', 74409)]

In [10]:
%%sql
select
    province_name_en,
    district_name_en,
    sum(interest) as total_interest
from customer_interest
group by province_name_en, district_name_en
order by province_name_en, total_interest desc;

 * postgresql://postgres:***@localhost:5432/customerinterest
77 rows affected.


[('Bangkok', 'SAI MAI', 81284),
 ('Bangkok', 'KHLONG SAM WA', 57990),
 ('Bangkok', 'LAT KRABANG', 31204),
 ('Bangkok', 'NONG CHOK', 22688),
 ('Bangkok', 'PRAWET', 21968),
 ('Bangkok', 'BANG KHAE', 21872),
 ('Bangkok', 'SAPHAN SUNG', 20259),
 ('Bangkok', 'KHAN NA YAO', 19736),
 ('Bangkok', 'BANG KHEN', 18813),
 ('Bangkok', 'THAWI WATTHANA', 17533),
 ('Bangkok', 'DON MUEANG', 15845),
 ('Bangkok', 'THUNG KHRU', 14594),
 ('Bangkok', 'LAT PHRAO', 13893),
 ('Bangkok', 'CHATUCHAK', 13711),
 ('Bangkok', 'BANG KHUN THIAN', 13004),
 ('Bangkok', 'MIN BURI', 11641),
 ('Bangkok', 'SUAN LUANG', 10302),
 ('Bangkok', 'NONG KHAEM', 9777),
 ('Bangkok', 'BUENG KUM', 9750),
 ('Bangkok', 'WANG THONGLANG', 9020),
 ('Bangkok', 'VADHANA', 8884),
 ('Bangkok', 'BANG NA', 7386),
 ('Bangkok', 'BANG KAPI', 7097),
 ('Bangkok', 'YAN NAWA', 6344),
 ('Bangkok', 'LAK SI', 6331),
 ('Bangkok', 'CHOM THONG', 6119),
 ('Bangkok', 'TALING CHAN', 5589),
 ('Bangkok', 'HUAI KHWANG', 5537),
 ('Bangkok', 'BANG SUE', 5520),
 ('Bangkok', 'PHRA KHANONG', 5381),
 ('Bangkok', 'BANGKOK NOI', 4517),
 ('Bangkok', 'BANG RAK', 4134),
 ('Bangkok', 'BANG PHLAT', 3910),
 ('Bangkok', 'PHAYA THAI', 3885),
 ('Bangkok', 'BANG KHO LAEM', 3560),
 ('Bangkok', 'THON BURI', 2967),
 ('Bangkok', 'PATHUM WAN', 2956),
 ('Bangkok', 'PHASI CHAROEN', 2929),
 ('Bangkok', 'KHLONG TOEI', 2703),
 ('Bangkok', 'DIN DAENG', 2362),
 ('Bangkok', 'BANG BON', 2362),
 ('Bangkok', 'BANGKOK YAI', 2150),
 ('Bangkok', 'RAT BURANA', 2081),
 ('Bangkok', 'SATHON', 1572),
 ('Bangkok', 'KHLONG SAN', 1432),
 ('Bangkok', 'RATCHATHEWI', 917),
 ('Bangkok', 'DUSIT', 209),
 ('Bangkok', 'POM PRAP SATTRU PHAI', 165),
 ('Bangkok', 'PHRA NAKHON', 64),
 ('Nakhon Pathom', 'SAM PHRAN', 72686),
 ('Nakhon Pathom', 'MUEANG NAKHON PATHOM', 30511),
 ('Nakhon Pathom', 'PHUTTHAMONTHON', 4304),
 ('Nakhon Pathom', 'NAKHON CHAI SI', 1594),
 ('Nakhon Pathom', 'KAMPHAENG SAEN', 1130),
 ('Nakhon Pathom', 'BANG LEN', 64),
 ('Nonthaburi', 'BANG BUA THONG', 80541),
 ('Nonthaburi', 'BANG YAI', 58845),
 ('Nonthaburi', 'SAI NOI', 45707),
 ('Nonthaburi', 'BANG KRUAI', 34297),
 ('Nonthaburi', 'MUEANG NONTHABURI', 21401),
 ('Nonthaburi', 'PAK KRET', 18212),
 ('Pathum Thani', 'KHLONG LUANG', 80874),
 ('Pathum Thani', 'LAM LUK KA', 52989),
 ('Pathum Thani', 'MUEANG PATHUM THANI', 43868),
 ('Pathum Thani', 'THANYABURI', 34411),
 ('Pathum Thani', 'SAM KHOK', 9751),
 ('Pathum Thani', 'LAT LUM KAEO', 6522),
 ('Pathum Thani', 'NONG SUEA', 651),
 ('Samut Prakan', 'BANG PHLI', 76938),
 ('Samut Prakan', 'MUEANG SAMUT PRAKAN', 42647),
 ('Samut Prakan', 'PHRA SAMUT CHEDI', 39566),
 ('Samut Prakan', 'BANG BO', 38491),
 ('Samut Prakan', 'BANG SAO THONG', 20480),
 ('Samut Prakan', 'PHRA PRADAENG', 5546),
 ('Samut Sakhon', 'MUEANG SAMUT SAKHON', 50026),
 ('Samut Sakhon', 'KRATHUM BAEN', 17653),
 ('Samut Sakhon', 'BAN PHAEO', 6730)]

2)  Yearly customer interest trend, How has customer interest evolved over the years?

    To see how customer preferences and behaviors have changed over times, revealing period of growth, stagnation, or decline in interest.

In [11]:
%%sql
select
    transaction_year,
    sum(interest) as total_interest
from customer_interest
group by transaction_year
order by total_interest;

 * postgresql://postgres:***@localhost:5432/customerinterest
5 rows affected.


[('2023', 83),
 ('2022', 159055),
 ('2021', 184231),
 ('2019', 542567),
 ('2020', 554446)]

3) Monthly customer interest trend, Are there specific months with increased customer interest?

In [12]:
%%sql
select 
    transaction_month,
    sum(interest) as total_interest
from customer_interest
group by transaction_month
order by total_interest desc;

 * postgresql://postgres:***@localhost:5432/customerinterest
12 rows affected.


[('3', 246761),
 ('7', 216103),
 ('6', 128525),
 ('1', 120320),
 ('5', 106573),
 ('10', 105062),
 ('9', 102096),
 ('2', 100874),
 ('11', 96170),
 ('12', 84150),
 ('4', 74601),
 ('8', 59147)]

4) Most popular property types by interest, Which property types are the most popular among customers across different locations?

In [13]:
%%sql
select
    propertytype_name_en,
    sum(interest) as total_interest
from customer_interest
group by propertytype_name_en
order by total_interest desc;

 * postgresql://postgres:***@localhost:5432/customerinterest
11 rows affected.


[('Detached House', 511872),
 ('Townhome', 479508),
 ('Twin house', 258806),
 ('Condo', 140680),
 ('Commercial', 33995),
 ('Home office', 14194),
 ('Apartment', 429),
 ('Warehouse / Factory', 355),
 ('Land', 353),
 ('Hotel', 118),
 ('Office Space', 72)]

In [14]:
%%sql
select
    province_name_en,
    propertytype_name_en,
    sum(interest) as total_interest
from customer_interest
group by province_name_en, propertytype_name_en
order by province_name_en, total_interest desc;

 * postgresql://postgres:***@localhost:5432/customerinterest
43 rows affected.


[('Bangkok', 'Detached House', 193717),
 ('Bangkok', 'Townhome', 171449),
 ('Bangkok', 'Condo', 101122),
 ('Bangkok', 'Twin house', 63919),
 ('Bangkok', 'Home office', 7601),
 ('Bangkok', 'Commercial', 5167),
 ('Bangkok', 'Apartment', 429),
 ('Bangkok', 'Land', 353),
 ('Bangkok', 'Hotel', 118),
 ('Bangkok', 'Office Space', 72),
 ('Nakhon Pathom', 'Detached House', 43426),
 ('Nakhon Pathom', 'Townhome', 35101),
 ('Nakhon Pathom', 'Twin house', 26922),
 ('Nakhon Pathom', 'Commercial', 2387),
 ('Nakhon Pathom', 'Condo', 2020),
 ('Nakhon Pathom', 'Home office', 433),
 ('Nonthaburi', 'Detached House', 99810),
 ('Nonthaburi', 'Townhome', 80335),
 ('Nonthaburi', 'Twin house', 52146),
 ('Nonthaburi', 'Condo', 12386),
 ('Nonthaburi', 'Commercial', 12077),
 ('Nonthaburi', 'Home office', 2249),
 ('Pathum Thani', 'Detached House', 84105),
 ('Pathum Thani', 'Townhome', 70927),
 ('Pathum Thani', 'Twin house', 51821),
 ('Pathum Thani', 'Condo', 14321),
 ('Pathum Thani', 'Commercial', 6105),
 ('Pathum Thani', 'Home office', 1698),
 ('Pathum Thani', 'Warehouse / Factory', 89),
 ('Samut Prakan', 'Townhome', 92116),
 ('Samut Prakan', 'Detached House', 68099),
 ('Samut Prakan', 'Twin house', 46181),
 ('Samut Prakan', 'Condo', 10172),
 ('Samut Prakan', 'Commercial', 4919),
 ('Samut Prakan', 'Home office', 1993),
 ('Samut Prakan', 'Warehouse / Factory', 188),
 ('Samut Sakhon', 'Townhome', 29580),
 ('Samut Sakhon', 'Detached House', 22715),
 ('Samut Sakhon', 'Twin house', 17817),
 ('Samut Sakhon', 'Commercial', 3340),
 ('Samut Sakhon', 'Condo', 659),
 ('Samut Sakhon', 'Home office', 220),
 ('Samut Sakhon', 'Warehouse / Factory', 78)]

5) Correlation between volume of projects and customer interest, Is there a correlation between the volume of projects and customer interest?

In [15]:
%%sql
select 
    province_name_en,
    propertytype_name_en,
    sum(volume_of_project) as total_volume_of_project,
    sum(interest) as total_interest
from customer_interest
group by province_name_en, propertytype_name_en
order by province_name_en, total_volume_of_project desc, total_interest desc;


 * postgresql://postgres:***@localhost:5432/customerinterest
43 rows affected.


[('Bangkok', 'Condo', 16240, 101122),
 ('Bangkok', 'Detached House', 13502, 193717),
 ('Bangkok', 'Townhome', 12253, 171449),
 ('Bangkok', 'Twin house', 3465, 63919),
 ('Bangkok', 'Home office', 1531, 7601),
 ('Bangkok', 'Commercial', 610, 5167),
 ('Bangkok', 'Apartment', 74, 429),
 ('Bangkok', 'Hotel', 36, 118),
 ('Bangkok', 'Land', 34, 353),
 ('Bangkok', 'Office Space', 32, 72),
 ('Nakhon Pathom', 'Detached House', 1754, 43426),
 ('Nakhon Pathom', 'Townhome', 1429, 35101),
 ('Nakhon Pathom', 'Twin house', 855, 26922),
 ('Nakhon Pathom', 'Commercial', 272, 2387),
 ('Nakhon Pathom', 'Condo', 239, 2020),
 ('Nakhon Pathom', 'Home office', 89, 433),
 ('Nonthaburi', 'Detached House', 5217, 99810),
 ('Nonthaburi', 'Townhome', 4253, 80335),
 ('Nonthaburi', 'Twin house', 2462, 52146),
 ('Nonthaburi', 'Condo', 1309, 12386),
 ('Nonthaburi', 'Commercial', 413, 12077),
 ('Nonthaburi', 'Home office', 169, 2249),
 ('Pathum Thani', 'Detached House', 4763, 84105),
 ('Pathum Thani', 'Townhome', 4618, 70927),
 ('Pathum Thani', 'Twin house', 3182, 51821),
 ('Pathum Thani', 'Condo', 1209, 14321),
 ('Pathum Thani', 'Commercial', 525, 6105),
 ('Pathum Thani', 'Home office', 330, 1698),
 ('Pathum Thani', 'Warehouse / Factory', 29, 89),
 ('Samut Prakan', 'Townhome', 5260, 92116),
 ('Samut Prakan', 'Detached House', 4671, 68099),
 ('Samut Prakan', 'Twin house', 2663, 46181),
 ('Samut Prakan', 'Condo', 1155, 10172),
 ('Samut Prakan', 'Commercial', 290, 4919),
 ('Samut Prakan', 'Home office', 236, 1993),
 ('Samut Prakan', 'Warehouse / Factory', 50, 188),
 ('Samut Sakhon', 'Townhome', 1882, 29580),
 ('Samut Sakhon', 'Detached House', 1301, 22715),
 ('Samut Sakhon', 'Twin house', 1029, 17817),
 ('Samut Sakhon', 'Commercial', 298, 3340),
 ('Samut Sakhon', 'Condo', 48, 659),
 ('Samut Sakhon', 'Home office', 40, 220),
 ('Samut Sakhon', 'Warehouse / Factory', 21, 78)]

6) Monthly trend of interest for each province.

In [16]:
%%sql
with monthlytrend as (
    select
        province_name_en,
        concat(transaction_year, '-', lpad(transaction_month, 2, '0')) as month_year,
        sum(interest) as monthly_interest
    from customer_interest
    group by province_name_en, transaction_year, transaction_month
)

select *
from monthlytrend
order by province_name_en, month_year;

 * postgresql://postgres:***@localhost:5432/customerinterest
244 rows affected.


[('Bangkok', '2019-01', 2046),
 ('Bangkok', '2019-02', 6985),
 ('Bangkok', '2019-03', 10500),
 ('Bangkok', '2019-04', 2310),
 ('Bangkok', '2019-05', 23210),
 ('Bangkok', '2019-06', 67941),
 ('Bangkok', '2019-07', 66134),
 ('Bangkok', '2019-08', 3205),
 ('Bangkok', '2019-09', 16865),
 ('Bangkok', '2019-10', 4238),
 ('Bangkok', '2019-11', 16744),
 ('Bangkok', '2019-12', 17295),
 ('Bangkok', '2020-01', 38455),
 ('Bangkok', '2020-02', 23054),
 ('Bangkok', '2020-03', 29543),
 ('Bangkok', '2020-04', 11795),
 ('Bangkok', '2020-05', 16236),
 ('Bangkok', '2020-06', 10847),
 ('Bangkok', '2020-07', 4888),
 ('Bangkok', '2020-08', 2367),
 ('Bangkok', '2020-09', 2853),
 ('Bangkok', '2020-10', 2872),
 ('Bangkok', '2020-11', 5894),
 ('Bangkok', '2020-12', 40608),
 ('Bangkok', '2021-01', 78),
 ('Bangkok', '2021-03', 9813),
 ('Bangkok', '2021-04', 8319),
 ('Bangkok', '2021-05', 3252),
 ('Bangkok', '2021-06', 1959),
 ('Bangkok', '2021-07', 1548),
 ('Bangkok', '2021-08', 4768),
 ('Bangkok', '2021-09', 12130),
 ('Bangkok', '2021-10', 9759),
 ('Bangkok', '2021-11', 3741),
 ('Bangkok', '2021-12', 6799),
 ('Bangkok', '2022-01', 18001),
 ('Bangkok', '2022-02', 3028),
 ('Bangkok', '2022-03', 7000),
 ('Bangkok', '2022-04', 4915),
 ('Bangkok', '2022-05', 4856),
 ('Bangkok', '2022-06', 3781),
 ('Bangkok', '2022-07', 3069),
 ('Bangkok', '2022-08', 3364),
 ('Bangkok', '2022-09', 4214),
 ('Bangkok', '2022-10', 1585),
 ('Bangkok', '2022-11', 672),
 ('Bangkok', '2022-12', 398),
 ('Bangkok', '2023-01', 13),
 ('Nakhon Pathom', '2019-03', 2907),
 ('Nakhon Pathom', '2019-04', 864),
 ('Nakhon Pathom', '2019-06', 2024),
 ('Nakhon Pathom', '2019-07', 7620),
 ('Nakhon Pathom', '2019-09', 4357),
 ('Nakhon Pathom', '2019-12', 1658),
 ('Nakhon Pathom', '2020-01', 4118),
 ('Nakhon Pathom', '2020-03', 40730),
 ('Nakhon Pathom', '2020-05', 2069),
 ('Nakhon Pathom', '2020-06', 2876),
 ('Nakhon Pathom', '2020-07', 509),
 ('Nakhon Pathom', '2020-08', 700),
 ('Nakhon Pathom', '2020-09', 1562),
 ('Nakhon Pathom', '2020-11', 2177),
 ('Nakhon Pathom', '2021-02', 470),
 ('Nakhon Pathom', '2021-03', 6457),
 ('Nakhon Pathom', '2021-05', 202),
 ('Nakhon Pathom', '2021-07', 175),
 ('Nakhon Pathom', '2021-08', 581),
 ('Nakhon Pathom', '2021-09', 907),
 ('Nakhon Pathom', '2021-10', 5812),
 ('Nakhon Pathom', '2021-12', 3182),
 ('Nakhon Pathom', '2022-01', 4852),
 ('Nakhon Pathom', '2022-02', 7158),
 ('Nakhon Pathom', '2022-03', 510),
 ('Nakhon Pathom', '2022-05', 582),
 ('Nakhon Pathom', '2022-06', 1590),
 ('Nakhon Pathom', '2022-07', 2095),
 ('Nakhon Pathom', '2022-09', 1072),
 ('Nakhon Pathom', '2022-11', 12),
 ('Nakhon Pathom', '2022-12', 451),
 ('Nakhon Pathom', '2023-01', 10),
 ('Nonthaburi', '2019-02', 20279),
 ('Nonthaburi', '2019-03', 2659),
 ('Nonthaburi', '2019-04', 759),
 ('Nonthaburi', '2019-05', 1094),
 ('Nonthaburi', '2019-06', 42),
 ('Nonthaburi', '2019-07', 26880),
 ('Nonthaburi', '2019-08', 9574),
 ('Nonthaburi', '2019-09', 25779),
 ('Nonthaburi', '2019-10', 52396),
 ('Nonthaburi', '2019-11', 4093),
 ('Nonthaburi', '2020-01', 17648),
 ('Nonthaburi', '2020-02', 5227),
 ('Nonthaburi', '2020-03', 9876),
 ('Nonthaburi', '2020-04', 4687),
 ('Nonthaburi', '2020-05', 8485),
 ('Nonthaburi', '2020-06', 3319),
 ('Nonthaburi', '2020-07', 11695),
 ('Nonthaburi', '2020-08', 1263),
 ('Nonthaburi', '2020-09', 1228),
 ('Nonthaburi', '2020-10', 495),
 ('Nonthaburi', '2020-11', 3390),
 ('Nonthaburi', '2021-02', 232),
 ('Nonthaburi', '2021-03', 8960),
 ('Nonthaburi', '2021-04', 358),
 ('Nonthaburi', '2021-05', 2003),
 ('Nonthaburi', '2021-06', 59),
 ('Nonthaburi', '2021-07', 6897),
 ('Nonthaburi', '2021-08', 923),
 ('Nonthaburi', '2021-09', 1563),
 ('Nonthaburi', '2021-10', 3154),
 ('Nonthaburi', '2021-11', 1844),
 ('Nonthaburi', '2021-12', 1677),
 ('Nonthaburi', '2022-01', 1659),
 ('Nonthaburi', '2022-02', 480),
 ('Nonthaburi', '2022-03', 2038),
 ('Nonthaburi', '2022-04', 2359),
 ('Nonthaburi', '2022-05', 4240),
 ('Nonthaburi', '2022-06', 1865),
 ('Nonthaburi', '2022-07', 3123),
 ('Nonthab

7) Top property type for each province.

In [17]:
%%sql
with propertyranking as (
    select
        province_name_en,
        propertytype_name_en,
        sum(interest) as total_interest,
        rank() over(partition by province_name_en order by sum(interest) desc) as rank
    from customer_interest
    group by province_name_en, propertytype_name_en
)

select *
from propertyranking 
where rank = 1;

 * postgresql://postgres:***@localhost:5432/customerinterest
6 rows affected.


[('Bangkok', 'Detached House', 193717, 1),
 ('Nakhon Pathom', 'Detached House', 43426, 1),
 ('Nonthaburi', 'Detached House', 99810, 1),
 ('Pathum Thani', 'Detached House', 84105, 1),
 ('Samut Prakan', 'Townhome', 92116, 1),
 ('Samut Sakhon', 'Townhome', 29580, 1)]

8) Average interest per volume of project.

In [18]:
%%sql
select 
    province_name_en,
    district_name_en,
    sum(interest)/nullif(sum(volume_of_project), 0) as avg_interest_per_volume
from customer_interest
group by province_name_en,district_name_en
order by avg_interest_per_volume desc
limit 10;

 * postgresql://postgres:***@localhost:5432/customerinterest
10 rows affected.


[('Nonthaburi', 'SAI NOI', 36),
 ('Samut Sakhon', 'BAN PHAEO', 36),
 ('Nakhon Pathom', 'SAM PHRAN', 27),
 ('Pathum Thani', 'SAM KHOK', 25),
 ('Bangkok', 'SAI MAI', 25),
 ('Nonthaburi', 'BANG YAI', 24),
 ('Nonthaburi', 'BANG BUA THONG', 22),
 ('Samut Prakan', 'BANG BO', 22),
 ('Bangkok', 'KHLONG SAM WA', 21),
 ('Pathum Thani', 'THANYABURI', 21)]